ServiceBus/ServiceBus/ProductService.svc.cs

```diff
@@ -128,5 +130,122 @@ public Result AddProduct(string name, double price, Guid guid)
+                    };
+
+                    if (!String.IsNullOrWhiteSpace(name) && name.Length <= 50 && name != editableProduct.Name)
+                    {
+                        changesName = true;
+                    }
+
+                    if (price != editableProduct.Price && price >= 0)
+                    {
+                        changesPrice = true;
+                    }
+
+                    if (changesName || changesPrice)
+                    {
+                        using (var context = new ServiceBusDatabaseEntities())
+                        {
+                            //TODO In case of more columns find a way to simplify this! 21.10.2016
+                            if (changesName)
+                            {
+                                editableProduct.Name = name;
+                            }
+                            if (changesPrice)
+                            {
+                                editableProduct.Price = price;
+                            }
+                            //context.Products.Attach(editableProduct);
+                            //Since there are only few columns I would prefer this
+                            context.Entry(editableProduct).State = EntityState.Modified;
+                            context.SaveChanges();
+                        }
+
+                        return new SharedLibs.DataContracts.Product()
+                        {
+                            ID = editableProduct.Id,
+                            Name = editableProduct.Name,
+                            Price = editableProduct.Price,
+                            Result = Result.Success()
+                        };
+                    }
+                    else
+                    {
+                        originalProduct.Result = Result.Warning("This product was not modified.");
+                        return originalProduct;
+                    }
+                }
+                else
+                {
+                    originalProduct.Result = Result.Error("This product was not found.");
+                    return originalProduct;
+                }
+
+            }
+            catch (Exception exception)
+            {
+                return new SharedLibs.DataContracts.Product()
+                {
+                    Result =
+                        Result.FatalFormat("ProductService.EditProduct exception has occured : {0}", exception.Message)
+                };
+            }
+        }
+
+        /// <summary>
+        /// Delete product item from datasource
+        /// </summary>
+        /// <param name="guid">ID of a product</param>
+        /// <returns>Result object</returns>
+        public Result DeleteProduct(Guid guid)
+        {
+            try
+            {   //TODO : With more complex table structure this should do something like "UPDATE db.Product SET isValid = 0 where db.Product.ID = guid and isValid = 1". Now it is fine as is.

```

Comment:
Yes, in longterm thinking, invalidating products is a better solution than deleting it (solving legacy issues in the future).